### PR TITLE
Pro 6605 mobile preview

### PR DESCRIPTION
**Pull request by BoDonkey**
## Summary

*Summarize the changes briefly, including which issue/ticket this resolves. If it closes an existing Github issue, include "Closes #[issue number]"*
This PR documents the new mobile preview in both the pages guide and the asset module reference. Closes PRO-6605.
## What are the specific steps to test this change?

*For example:*
> 1. Run the website and log in as an admin
> 2. Open a piece manager modal and select several pieces
> 3. Click the "Archive" button on the top left of the manager and confirm that it should proceed
> 4. Check that all pieces have been archived properly

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [ ] New feature
- [ ] Refactor
- [X] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [X] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [ ] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: docs/guide/pages.md [hunk 1]
```diff
@@ -244,3 +244,53 @@ With that available data, we could construct navigation for the website header u
 ```
 
 This is looping over the home page's child pages, printing their URLs and titles into links. This is simply one way to add navigation using the page data in templates.
+
+## Device breakpoint previews
+
+Pages can be enabled to emulate devices, like phones and tablets, making it easy to see how your content will look at different screen sizes.
+
+![Screen shot of mobile preview with ecommerce](../images/mobile-preview.png)
+When device preview mode is enabled, you’ll see icons in the admin bar for each of the breakpoints (sizes) that have been set through the `devicePreviewMode` of the `@apostrophecms/asset` module. Clicking on an icon will cause your page content to display in a container that matches the specific device size, so you can quickly check how the layout and styles respond. The displayed content will be fully editable, so that you can see how any new content is impacted by the current media queries. You can go back to editing in the full browser screen by clicking on the currently selected preview icon.
```
[the remark this answers is not in the document]
Comment (boutell) [reply docs/guide/pages.md:253]: We are changing the option name.

File: docs/guide/pages.md [hunk 2]
```diff
@@ -244,3 +244,53 @@ With that available data, we could construct navigation for the website header u
...
+
+Custom styles you’ve added with CSS media queries are converted by the `asset` module to container queries. Note that this only works with styles added to stylesheets, it doesn't work with styles outside the build path, like styles added directly to the template. Keep in mind that while most media queries will translate well, there are some differences when using container queries that might not be fully reflected in the preview. For example, you can not add `min-height` or `rotation` query parameters to containers.
+
+To enable preview, you can pass options to the `@apostrophecms/asset` module at project level.
```
[the remark this answers is not in the document]
Comment (boutell) [reply docs/guide/pages.md:257]: "It is enabled by default, but if you want to change the configuration..."

File: docs/reference/modules/asset.md [hunk 3]
```diff
@@ -77,6 +78,105 @@ modules/@apostrophecms/asset/index.js
 
 To split files within a single `ui/src` folder into multiple bundles, assign each file separately with a property:value pair for each file.
 
+### `devicePreviewMode`
+The `devicePreviewMode` is disabled by default and when activated adds icons to the admin-bar for each of the breakpoints specified in the `screens` object. Clicking on these icons will cause the markup to display as a CSS container of the specified dimensions. Any styling assets handled by the apostrophe build process will be applied as container queries in this display. This means that CSS breakpoint styling added through `<style>` blocks in the template will not change in response to the altered display size. There are also some standard CSS breakpoint declarations that cannot be directly converted into container queries, particularly those targeting the viewport height or those relying on global viewport conditions. You will receive console notifications when declarations that can't be converted are encountered if the `debug` property is set to `true`.
```
Comment (boutell) [inline docs/reference/modules/asset.md:82]: * Enabled by default
* We need to explain first that we are transpiling to container queries only for preview purposes and we do not alter them outside of preview

**Review by boutell — CHANGES_REQUESTED** [review]
ACHTUNG: most of my points here are subject to Harouna's approval of my PR on the actual code in the morning.

File: docs/guide/pages.md [hunk 4]
```diff
@@ -244,3 +244,53 @@ With that available data, we could construct navigation for the website header u
...
+When device preview mode is enabled, you’ll see icons in the admin bar for each of the breakpoints (sizes) that have been set through the `devicePreviewMode` of the `@apostrophecms/asset` module. Clicking on an icon will cause your page content to display in a container that matches the specific device size, so you can quickly check how the layout and styles respond. The displayed content will be fully editable, so that you can see how any new content is impacted by the current media queries. You can go back to editing in the full browser screen by clicking on the currently selected preview icon.
+
+Custom styles you’ve added with CSS media queries are converted by the `asset` module to container queries. Note that this only works with styles added to stylesheets, it doesn't work with styles outside the build path, like styles added directly to the template. Keep in mind that while most media queries will translate well, there are some differences when using container queries that might not be fully reflected in the preview. For example, you can not add `min-height` or `rotation` query parameters to containers.
```
[the remark this answers is not in the document]
Comment (BoDonkey) [reply docs/guide/pages.md:255]: Changed and link added.

File: docs/guide/pages.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (BoDonkey) [reply docs/guide/pages.md:257]: Changed.

File: docs/reference/modules/asset.md (hunk 3 above)
Comment (BoDonkey) [reply docs/reference/modules/asset.md:82]: Changed.

**Review by boutell — APPROVED** [review]

**Pull request merged**